Source/ACE.Server/Command/Handlers/AdminCommands.cs

```diff
@@ -1584,59 +1584,268 @@ public static void HandleFumble(Session session, params string[] parameters)
                 playerAttr.Ranks = 190u;
                 playerAttr.ExperienceSpent = 4019438644u;
-                session.Player.Session.Network.EnqueueSend(new GameMessagePrivateUpdateAttribute(session.Player, a.Key, playerAttr.Ranks, playerAttr.StartingValue, playerAttr.ExperienceSpent));
+                currentPlayer.Session.Network.EnqueueSend(new GameMessagePrivateUpdateAttribute(currentPlayer, a.Key, playerAttr.Ranks, playerAttr.StartingValue, playerAttr.ExperienceSpent));
             }
 
-            session.Player.SetMaxVitals();
+            currentPlayer.SetMaxVitals();
 
-            foreach (var v in session.Player.Vitals)
+            foreach (var v in currentPlayer.Vitals)
             {
-                var playerVital = session.Player.Vitals[v.Key];
+                var playerVital = currentPlayer.Vitals[v.Key];
                 playerVital.Ranks = 196u;
                 playerVital.ExperienceSpent = 4285430197u;
                 // my OCD will not let health/stam not be equal due to the endurance calc
                 playerVital.StartingValue = (v.Key == PropertyAttribute2nd.MaxHealth) ? 94803u : 89804u;
-                session.Player.Session.Network.EnqueueSend(new GameMessagePrivateUpdateVital(session.Player, v.Key, playerVital.Ranks, playerVital.StartingValue, playerVital.ExperienceSpent, playerVital.Current));
+                currentPlayer.Session.Network.EnqueueSend(new GameMessagePrivateUpdateVital(currentPlayer, v.Key, playerVital.Ranks, playerVital.StartingValue, playerVital.ExperienceSpent, playerVital.Current));
             }
 
-            session.Player.SetMaxVitals();
-
-            session.Player.ChangesDetected = true;
+            currentPlayer.PlayParticleEffect(PlayScript.LevelUp, currentPlayer.Guid);
+            currentPlayer.PlayParticleEffect(PlayScript.BaelZharonSmite, currentPlayer.Guid);
 
-            session.Player.PlayParticleEffect(PlayScript.LevelUp, session.Player.Guid);
-            session.Player.PlayParticleEffect(PlayScript.BaelZharonSmite, session.Player.Guid);
+            currentPlayer.SetMaxVitals();
 
             ChatPacket.SendServerMessage(session, "You are now a god!!!", ChatMessageType.Broadcast);
+        }
+
+        // ungod
+        [CommandHandler("ungod", AccessLevel.Sentinel, CommandHandlerFlag.RequiresWorld, 0,
+            "Returns character to a mortal state.",
+            "Sets attributes and skills back to the values they were when you became a god.\n"
+            + "If the command fails to revert your state it will default to godmode.\n"
+            + "In the event of failure, contact a server administrator to sort it out.")]
+        public static void HandleUngod(Session session, params string[] parameters)
+        {
+            // @ungod - Returns skills and attributues to pre-god levels.
+            Player currentPlayer = session.Player;
+            Biota biota = DatabaseManager.Shard.GetBiota(currentPlayer.Guid.Full);
```

Comment:
again, shouldn't be any reason to fetch a copy of the raw Biota here. session.Player already has the master reference to the raw biota if needed, and ideally session.Player gamevars should just be dealt with directly

Reply:
right, no need to hit db. just grabbing session.player.biota